sql/src/main/java/io/crate/metadata/information/InformationKeyColumnUsageTableInfo.java

```diff
@@ -0,0 +1,102 @@
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.metadata.information;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSortedMap;
+import io.crate.metadata.*;
+import io.crate.metadata.expressions.RowCollectExpressionFactory;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import org.elasticsearch.cluster.service.ClusterService;
+
+public class InformationKeyColumnUsageTableInfo extends InformationTableInfo {
+
+    public static final String NAME = "key_column_usage";
+    public static final TableIdent IDENT = new TableIdent(InformationSchemaInfo.NAME, NAME);
+
+    public static class Columns {
+        static final ColumnIdent CONSTRAINT_CATALOG = new ColumnIdent("constraint_catalog");
+        static final ColumnIdent CONSTRAINT_SCHEMA = new ColumnIdent("constraint_schema");
+        static final ColumnIdent CONSTRAINT_NAME = new ColumnIdent("constraint_name");
+        static final ColumnIdent TABLE_CATALOG = new ColumnIdent("table_catalog");
+        static final ColumnIdent TABLE_SCHEMA = new ColumnIdent("table_schema");
+        static final ColumnIdent TABLE_NAME = new ColumnIdent("table_name");
+        static final ColumnIdent COLUMN_NAME = new ColumnIdent("column_name");
+        static final ColumnIdent ORDINAL_POSITION = new ColumnIdent("ordinal_position");
+    }
+
+    public static class References {
+        static final Reference CONSTRAINT_CATALOG = createRef(Columns.CONSTRAINT_CATALOG, DataTypes.STRING);
+        static final Reference CONSTRAINT_SCHEMA = createRef(Columns.CONSTRAINT_SCHEMA, DataTypes.STRING);
+        static final Reference CONSTRAINT_NAME = createRef(Columns.CONSTRAINT_NAME, DataTypes.STRING);
+        static final Reference TABLE_CATALOG = createRef(Columns.TABLE_CATALOG, DataTypes.STRING);
+        static final Reference TABLE_SCHEMA = createRef(Columns.TABLE_SCHEMA, DataTypes.STRING);
+        static final Reference TABLE_NAME = createRef(Columns.TABLE_NAME, DataTypes.STRING);
+        static final Reference COLUMN_NAME = createRef(Columns.COLUMN_NAME, DataTypes.STRING);
+        static final Reference ORDINAL_POSITION = createRef(Columns.ORDINAL_POSITION, DataTypes.SHORT);
+    }
+
+    public static ImmutableMap<ColumnIdent, RowCollectExpressionFactory<KeyColumnUsageInfo>> expressions() {
+        return ImmutableMap.<ColumnIdent, RowCollectExpressionFactory<KeyColumnUsageInfo>>builder()
+            .put(Columns.CONSTRAINT_CATALOG,
+                () -> RowContextCollectorExpression.objToBytesRef(KeyColumnUsageInfo::constraintCatalog))
```

Comment:
you can use `Void` as a generic type instead of `KeyColumnUsageInfo` and replace above line with
```() -> RowContextCollectorExpression.objToBytesRef(r -> null))```
then you can remove the `KeyColumnUsageInfo` class completely as I don't think it makes sense if all there is just returning nulls.